packages/components/table/src/store/index.ts

```diff
@@ -55,10 +55,7 @@ function useStore<T>() {
       instance.store.updateTreeData(
         instance.store.states.defaultExpandAll.value
       )
-      if (unref(states.reserveSelection)) {
-        instance.store.assertRowKey()
-        instance.store.updateSelectionByRowKey()
-      } else {
+      if (!unref(states.reserveSelection)) {
```

Comment:
`assertRowKey` need to keep.

![image](https://github.com/user-attachments/assets/7a0c86f8-25d1-4b07-b0a9-e748699905d5)


```suggestion
      if (unref(states.reserveSelection)) {
        instance.store.assertRowKey()
      } else {
```